src/Pact/Native.hs

```diff
@@ -735,6 +751,16 @@ makeList g i [TLitInteger len,value] = case typeof value of
   Left ty -> evalError' i $ "make-list: invalid value type: " <> pretty ty
 makeList _ i as = argsError i as
 
+enumerate :: GasRNativeFun e
+enumerate g i as@[TLitInteger from', TLitInteger to', TLitInteger inc']
+  | inc' == 0 = argsError i as
+  | otherwise =
+    computeGas' g i (GMakeList $ succ ((to' - from') `div` inc')) $ return $ toTList tTyInteger def $ toTerm <$> enumFromThenTo from' (from' + inc') to'
+enumerate g i [TLitInteger from', TLitInteger to']
+  | from' <= to' = computeGas' g i (GMakeList $ succ (to' - from')) $ return $ toTList tTyInteger def $ toTerm <$> enumFromThenTo from' (from' + 1) to'
+  | otherwise = computeGas' g i (GMakeList $ succ (from' - to')) $ return $ toTList tTyInteger def $ toTerm <$> enumFromThenTo from' (from' - 1) to'
+enumerate _ i as = argsError i as
+
```

Comment:
Splitting these cases out and piping common functionality to an auxiliary function may be more legible - your choice:

```haskell
enumerate :: GasRNativeFun e 
enumerate g i as = case as of 
  [TLitInteger from', TLitInteger to', TLitInteger inc'] 
    | inc' <= 0 -> argsError i as
    | otherwise -> enumerateDef ((to' - from') `div` inc') to' from' (from' + inc')
  [TLitInteger from', TLitInteger to']
    | from' <= to' -> enumerateDef (to' - from') to' from' (from' + 1)
    -- to' < from' 
    | otherwise -> enumerateDef (from' - to') to' from' (from' - 1)
  _ -> argsError i as
  where 
    enumarateDef gasCost t f inc' = computeGas' g i (GMakeList $ succ gasCost) $ do 
      let lstTerm = toTerm $ enumFromThenTo f inc' t 
      return $ toTList tTyInteger def lstTerm